drivers/flash/flash_page_layout.c

```diff
@@ -72,6 +72,41 @@ size_t z_impl_flash_get_page_count(const struct device *dev)
 	return count;
 }
 
+int z_impl_flash_get_erase_region(const struct device *dev, off_t offset,
+				  struct flash_erase_region *region)
+{
+	if ((dev == NULL) || (region == NULL) || (offset < 0)) {
```

Comment:
We should assert here, (and we should use size_t for the offset, no reason to allow negative values at all :))
```
__ASSERT(dev != NULL, "dev arg must be provided");
...
```
We don't need to check this every time in production code :)

Reply:
Regarding the use of off_t, this is used troughout the flash driver, I agree it could be size_t, but would like to keep off_t until it is changed troughout.

Regarding the use of `__ASSERT`, I will change this.

Reply:
> We should assert here, (and we should use size_t for the offset, no reason to allow negative values at all :))
> 
> ```
> __ASSERT(dev != NULL, "dev arg must be provided");
> ...
> ```
> 
> We don't need to check this every time in production code :)

The `off_t` is kinda known issue as it has been designed for POSIX file functions and should allow you to seek both directions in file. `size_t` is actually used for sizingof() the largest object that could potentially exist on device, which is probably tied to device word bits.
On the other side we could, potentially, serve external devices with sizes that could not be stored in 32 bit, while running on 32 bit device.
The only solution I can see here is to basically define Zephyr storage specific type, where user would select range at compile time, and replace usage of `off_t` and `size_t` with it.